app/src/main/java/com/nutomic/syncthingandroid/service/RunConditionMonitor.java

```diff
@@ -118,6 +118,7 @@ private class BatteryReceiver extends BroadcastReceiver {
         public void onReceive(Context context, Intent intent) {
             if (Intent.ACTION_POWER_CONNECTED.equals(intent.getAction())
                     || Intent.ACTION_POWER_DISCONNECTED.equals(intent.getAction())) {
+                SystemClock.sleep(5000);
```

Comment:
I believe this is running on the main/UI thread, i.e. freezes the app for 5s - that's not acceptable. Kicking off a timer (assume the decision doesn't touch UI) or using some handler should work.

Reply:
Done, thanks for the reminder. Build environment is broken atm though, I can't test this change on a real device.